File: pong_serveur.js
const express = require("express");
const request = require("request");
const axios = require("axios");
const app = express();
const registry = process.env.REGISTRY || "http://localhost:8080";
const address = process.env.ADDRESS || "http://localhost";
const port = parseInt(process.env.PORT) || 5372;


axios.post(registry+"/sent-address/s2", { data: "5372" });

axios.get(registry+"/s1").then((port) => {
   console.log(port.data);
  axios.post(address + port.data + "/send-data", { data: "Ping" });
});

  app.post("/send-data", (req, res) => {
    // Envoi de la donnée à server 2
    res.send("Ping");
    console.log("Ping");
    axios.get(registry+"/s1").then((port) => {
     
      axios.post(address + port.data + "/send-data", {
        data: "Ping",
      });
    });
  });

app.listen(port, () => {
  console.log(`Server 2 listening at http://localhost:${port}`);
});

File: pong_serveur.py
const express = require('express');
const app = express();
const port = 3000;

app.get('/send-data', (req, res) => {
    const data = { message: "Hello from server 1" };

    // Envoi de la donnée à server 2
    request.post({
        url: 'http://localhost:3001/receive-data',
        form: data
    }, (err, httpResponse, body) => {
        if (err) {
            console.log(err);
            res.status(500).send('Error communicating with server 2');
        } else {
            console.log(body);
            res.send('Data sent to server 2');
        }
    });
});

app.listen(port, () => {
    console.log(`Server 1 listening on port ${port}`);
});
